pkg/info/info_test.go

```diff
@@ -0,0 +1,48 @@
+package info_test
+
+import (
+	"runtime/debug"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/backplane-cli/pkg/info"
+)
+
+// Mock implementation of debug.ReadBuildInfo
+func mockReadBuildInfoAvail() (*debug.BuildInfo, bool) {
```

Comment:
It's great that you program to an interface, and you may able to generate the mock class rather than manually define it.
ex : https://github.com/openshift/backplane-cli/blob/16993591c759db8286194361e6d26df4341606d8/Makefile#L118C2-L118C132
```
#step-1 
mockgen -destination=./pkg/info/mocks/infoServiceMock.go -package=mocks github.com/openshift/backplane-cli/pkg/info InfoService

#step-2 
make mock-gen

# step-3 
# return what-ever-value you need 
mockInfoService.EXPECT().GetBuildVersion().Return("v1.2.3").Times(1)
``` 
if you see any issue with above steps , pls ping me and I'm happy to help :)

Reply:
Thank you for the suggestion and detailed steps @samanthajayasinghe !

In the new commit, I added the mock-gen to generate a mock interface.

In the test cases, I would like to test the logic in function `GetVersion()` (renamed from `GetBuildVersion()` in the new commit). 

The `GetVersion()` will call `ReadBuildInfo()` to get the `buildInfo` from the golang build.  In normal case, `ReadBuildInfo` points to a library function `debug.ReadBuildInfo` (runtime/debug) directly. For testing, I point `ReadBuildInfo` to a self-defined function to mock the desired output. I didn't make an interface for `debug.ReadBuildInfo`.

Not sure if that's the best practice. Would like you hear your suggestions.  Thanks!